seedData.js

```diff
@@ -1,52 +1,36 @@
 const chalk = require('chalk');
 
+const genners = ['pop', 'rock', 'rap', 'r&b', 'punk', 'emo', 'classical', 'jazz', 'spoken', 'ambient'];
+
+const producers = ['dre', 'michael williams', 'zaytoven', 'bialystock unt bloom', 'stock boy'];
+
 let seedData = function() {
-  let completed = 0;
-  for (let i = 1; i <= 100; i++) {
-    let id = i;
-    relatedData.findTrack(id, (err, track) => {
-      if (err) {
-        console.log(chalk.red('No data: ', err));
-        i += 100;
-      } else {
-        relatedData.deleteTrack(id, (err, track) => {
-          if (err) {
-            console.log(chalk.red('Problem deleting for seed data: ', err));
-          } else {
-            completed++;
-            if (completed === 100) {
-              let seeded = 0;
-              for (let j = 1; j <= 100; j++) {
-                let info = {song_id: j};
-                info.plays = Math.floor(Math.random() * 11);
-                info.likes = Math.floor(Math.random() * 11);
-                info.reposts = Math.floor(Math.random() * 11);
-                info.related = [];
-                let similar = Math.floor(Math.random() * 4);
-                for (let k = 1; k <= similar; k++) {
-                  let num = Math.ceil(Math.random() * 100);
-                  info.related.push(num);
-                }
-                relatedData.saveTrack(info, (err, data) => {
-                  if (err) {
-                    console.log(chalk.red('Problem seeding data: ', err));
-                  } else {
-                    seeded++;
-                    console.log(chalk.blue(data));
-                    if (seeded === 100) {
-                      console.log(chalk.green('Seeding complete'));
-                      return;
-                    }
-                  }
-                })
-              }
-            }
-          }
-        })
-      }
-    })
-  }
+  return new Promise((resolve, reject) => {
+    drop();
+    resolve();
+  })
+  .then(() => {
+    for (let i = 1; i <= 100; i += 1) {
+      let current = {
+        song_id: i,
+        plays: Math.ceil(Math.random() * 10000),
+        likes: Math.ceil(Math.random() * 1000),
+        reposts: Math.ceil(Math.random() * 100),
+        comments: Math.ceil(Math.random() * 100),
+        genre: genners[Math.floor(Math.random() * genners.length)],
+        onPlaylists: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        producedBy: producers[Math.floor(Math.random() * producers.length)],
+        likedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        playedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+        repostedBy: [Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100), Math.ceil(Math.random() * 100)],
+      };
+      addTrack(current, () => {});
```

Comment:
for the big seed file & import w/ 10m, you probably want to bulk import, not use addTrack each time

Reply:
Yep, I'm doing something similar to you. Writing csv files and just importing those directly.